README.md

```diff
@@ -15,218 +15,380 @@ and understand how to run and work with Dgraph.
 
-We define a person object to represent a person and serialize it to a json mutation string. In this example, we are using the [JSON.NET](https://www.newtonsoft.com/json) library, but you can use any JSON serialization library you prefer.
+While you can construct a `Dgraph.Api.Mutation` object directly, it is easier to let `MutationBuilder` handle implementation details like converting `string` to `Google.Protobuf.ByteString`.
 
 ```c#
-using(var txn = client.NewTransaction()) {
-    var alice = new Person{ Name = "Alice" };
-    var json = JsonConvert.SerializeObject(alice);
-    
-    var transactionResult = await txn.Mutate(new RequestBuilder().WithMutations(new MutationBuilder{ SetJson = json }));
+using Dgraph;
+
+var transaction = dgraphClient.NewTransaction();
+var mutation = new MutationBuilder().SetJson("...");
+var response = await transaction.Mutate(mutation);
+if (response.IsFailed) {
+    // Handle errors
 }
 ```
 
-You can also set mutations using RDF format, if you so prefer, as seen below:
+If you want to commit a mutation without querying anything further, use `MutationBuilder.CommitNow` to indicate that the transaction must be immediately committed.
 
 ```c#
-var mutation = "_:alice <name> \"Alice\"";
-var transactionResult = await txn.Mutate(new RequestBuilder().WithMutations(new MutationBuilder{ SetNquads = mutation }));
+var mutation = new MutationBuilder().SetJson("...").CommitNow();
 ```
 
+Multiple mutations can be run in a single request using `ITransaction.Do` and `RequestBuilder`. To immediately commit the request, use `RequestBuilder.CommitNow`.
+
+```c#
+using Dgraph;
+
+var transaction = dgraphClient.NewTransaction();
+var request = new RequestBuilder().WithMutations(
+    new MutationBuilder().SetJson("..."),
+    new MutationBuilder().SetJson("..."),
+    new MutationBuilder().SetJson("..."),
+);
+var response = await transaction.Do(request);
+if (response.IsFailed) {
+    // Handle errors
+}
+```
+
+Keep in mind that if you do not use `RequestBuilder.CommitNow` or `MutationBuilder.CommitNow`, you will still need to manually commit the transaction using `ITransaction.Commit`.
+
 Check out the example in `source/Dgraph.tests.e2e/TransactionTest.cs`.
 
 ### Running a Query
 
-You can run a query by calling `Transaction.Query(string)`. You will need to pass in a
-GraphQL+- query string. If you want to pass an additional map of any variables that
-you might want to set in the query, call `Transaction.QueryWithVars(string, Dictionary<string,string>)` with
-the variables dictionary as the second argument.
-
-The response would contain the response string.
+You can run a query by calling `ITransaction.Query`. You will need to pass in a DQL query string. If you want to pass an additional map of any variables that you might want to set in the query, call `ITransaction.QueryWithVars` with the variables dictionary as the second argument.
```

Comment:
why has the name changed from Transaction to ITransaction?

Reply:
It's an interface. See my commend below about IDgraphClient.